syntax = "proto3";

package protos;



import "google/protobuf/empty.proto";
import "custom_types.proto";

service TransactionCategoryService {
  rpc GetAll (GetAllRequest) returns (GetAllTransactionCategoriesResponse);
  rpc GetById (GetTransactionCategoryByIdRequest) returns (TransactionCategoryModel);
}

message TransactionCategoryModel {
  int32 category_id = 1;
  string name = 2;
}

message GetAllTransactionCategoriesResponse {
  repeated TransactionCategoryModel transaction_categories = 1;
}

message GetTransactionCategoryByIdRequest {
  int32 category_id = 1;
}
